app/src/main/kotlin/com/wire/android/ui/home/newconversation/search/SearchPeopleViewModel.kt

```diff
@@ -1,80 +1,109 @@
+import com.wire.kalium.logic.feature.publicuser.SearchKnownUsersUseCase
+import com.wire.kalium.logic.feature.publicuser.SearchUserDirectoryUseCase
+import com.wire.kalium.logic.functional.Either
 import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.flowOn
+import kotlinx.coroutines.flow.onStart
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
 import javax.inject.Inject
 
 @HiltViewModel
-class SearchPeopleViewModel @Inject constructor() : ViewModel() {
-
-    var state: SearchPeopleState by mutableStateOf(
-        SearchPeopleState()
-    )
-
-    @Suppress("MagicNumber")
-    fun search(searchQuery: String) {
-        state = state.copy(
-            searchQuery = searchQuery,
-            contactSearchResult = buildList {
-                for (i in 1..5) {
-                    add(
-                        Contact(
-                            id = i,
-                            "test$i",
-                            label = "test$i",
-                            eventType = EventType.UnreadMention
-                        )
-                    )
-                }
-                for (i in 1..5) {
-                    add(
-                        Contact(
-                            id = i,
-                            "test$i",
-                            label = "test$i",
-                            eventType = EventType.MissedCall
-                        )
-                    )
-                }
-                for (i in 1..5) {
-                    add(
-                        Contact(
-                            id = i,
-                            "test$i",
-                            label = "test$i",
-                            eventType = EventType.UnreadMessage(i)
-                        )
-                    )
-                }
-            },
-            publicContactSearchResult = buildList {
-                for (i in 1..15) {
-                    add(
-                        ExternalContact(
-                            id = i,
-                            "test$i",
-                            label = "test$i",
-                        )
-                    )
-                }
-            },
-            federatedContactSearchResult = buildList {
-                for (i in 1..15) {
-                    add(
-                        ExternalContact(
-                            id = i,
-                            "test$i",
-                            label = "test$i",
-                        )
-                    )
-                }
-            }
+class SearchPeopleViewModel @Inject constructor(
+    private val searchKnownUsers: SearchKnownUsersUseCase,
+    private val searchPublicUsers: SearchUserDirectoryUseCase
+) : ViewModel() {
+    private companion object {
+        const val HARDCODED_TEST_DOMAIN = "staging.zinfra.io"
```

Comment:
I will add a TODO annotation here to remember mapping it out with the given backend configuration later on

Reply:
done